server/routes/activityRouter.js

```diff
@@ -22,27 +36,33 @@ activityRouter.route('/:householdId/:studentId')
     .post((req, res) => {
         Household.findById((req.params.householdId), (err, household) => {
             if (err) {
-                return res.status(404).send("could not find household")
+                return res.status(500).send(err)
+            } else if (!household) {
+                return res.status(404).send(new Error("Could not find household."))
             }
-            var student = household.students.find((student) => student.id === req.studentId);
-            if (student === undefined) {
-                return res.status(404).send("could not find student")
+            const student = household.students.id(req.studentId);
+            if (student === null) {
+                return res.status(404).send(new Error("Could not find student."))
             }
-            var activity = new Activity(req.body)
-            activity.save()
+            const activity = new Activity(req.body)
             student.amount_due += activity.amount
-            student.activities.insertOne(activity)
-            res.status(201).json(activity)
+            student.activities.addToSet(activity)
+            household.save((err) => {
+                if (err) {
+                    return res.status(500).json(err)
+                }
+                res.status(201).json(activity)
+            })
         })
     })
     .get((req, res) => {
         Household.findById((req.params.householdId), (err, household) => {
             if (err) {
-                return res.status(404).json("could not find household")
+                return res.status(404).json(new Error("Could not find household"))
             }
-            var student = household.students.find((student) => student.id === req.studentId);
-            if (student === undefined) {
-                return res.status(404).json("could not find student")
+            var student = household.students.id(req.studentId);
```

Comment:
what is students.id? is that the mongoose made id or the one you are trying to use from the schema?

Reply:
This is Mongoose's built-in function that searches a nested array for students by auto-assigned id.